### PR TITLE
Knx 2.0 generic dpt mapper

### PR DESCRIPTION
**Pull request by lewie**
Testcomment

### CODE REVIEW

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java [hunk 1]
```diff
@@ -61,175 +69,461 @@
...
  * @author Volker Daube
+ * @author Helmut Lehmeyer
+ *         generic DPT Mapper
  * @since 0.3.0
- *
  */
 public class KNXCoreTypeMapper implements KNXTypeMapper {
 
-    static private final Logger logger = LoggerFactory.getLogger(KNXCoreTypeMapper.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(KNXCoreTypeMapper.class);
```
Comment (kgoderis) [inline addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java]: We should in fact remove static here, and LOGGER should become again 'logger'

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java [hunk 2]
```diff
@@ -61,175 +69,461 @@
...
+         * 1.002: Boolean, values false true
+         * 1.013: Dim send-style, values start/stop cyclic
+         * 1.003: Enable, values disable enable
+         * 1.014: Input source, values fixed calculated
+         * 1.004: Ramp, values no ramp ramp
+         * 1.015: Reset, values no action reset
+         */
+        dptMainTypeMap.put(1, OnOffType.class);
+        /** Exceptions Datapoint Types "B1", Main number 1 */
         dptTypeMap.put(DPTXlatorBoolean.DPT_UPDOWN.getID(), UpDownType.class);
```
Comment (kgoderis) [inline addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java:140]: Can you explain a bit why we have these exceptions directly put into the dptTypeMap?

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java [hunk 3]
```diff
@@ -61,175 +69,461 @@
...
-        // Datapoint Types "B1", Main number 1
-        dptTypeMap.put(DPTXlatorBoolean.DPT_SWITCH.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_BOOL.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_ENABLE.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_RAMP.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_ALARM.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_BINARYVALUE.getID(), OnOffType.class);
-        dptTypeMap.put(DPTXlatorBoolean.DPT_STEP.getID(), OnOffType.class);
+        @SuppressWarnings("unused")
+        final List<Class<?>> xlators = Arrays.<Class<?>> asList(DPTXlator1BitControlled.class,
```
Comment (kgoderis) [inline addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java]: should this not be static?

**Review by kgoderis — CHANGES_REQUESTED** [review]

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java (hunk 1 above)
Comment (lewie) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java]: Thought about, but then some static function can not log, [like this.](https://github.com/kgoderis/openhab2/pull/9/files/e96c5042fbc8ccea702ac758a7b5815c2f3ed5b1#diff-0a73a0644601e4bd4e031b0907472063R936)

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java (hunk 2 above)
Comment (lewie) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java:140]: >dptMainTypeMap.put(1, OnOffType.class); 

responds generic inclusive (OnOffType). But there are datatypes that do not fit into that pattern(s). Exceptions which we have to put specially.

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java (hunk 3 above)
Comment (lewie) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java]: It is in a static block:

```

    static {

        @SuppressWarnings("unused")
        final List<Class<?>> xlators = Arrays.<Class<?>> asList(DPTXlator1BitControlled.class,
                DPTXlator2ByteFloat.class, DPTXlator2ByteUnsigned.class, DPTXlator3BitControlled.class,
                DPTXlator4ByteFloat.class, DPTXlator4ByteSigned.class, DPTXlator4ByteUnsigned.class,
...
```

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java (hunk 2 above)
Comment (lewie) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java:140]: ...here like *UpDownType*

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java [hunk 4]
```diff
@@ -61,175 +69,461 @@
...
+        // Example: dptTypeMap.put(DPTXlator4ByteFloat.DPT_ACCELERATION_ANGULAR.getID(), DecimalType.class);
+
+        /**
+         * MainType: 16
+         * 16.000: ASCII string, values
+         * 16.001: ISO-8859-1 string (Latin 1), values
+         */
+        dptMainTypeMap.put(16, StringType.class);
+        /** Exceptions Datapoint Types "String", Main number 16 */
         dptTypeMap.put(DPTXlatorString.DPT_STRING_8859_1.getID(), StringType.class);
```
Comment (kgoderis) [inline addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java]: @lewie Ok, I was in fact looking at this line before, and here we have StringTYpe for the maintype, and also String for the exceptions. That's why I could not figure out what you were doing. So, does that mean that we do need those exception here in class 16?

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java (hunk 3 above)
Comment (kgoderis) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java]: Ok missed that line #95

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java (hunk 1 above)
Comment (lewie) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java]: Is a private static "LOGGER" ok for you in this class?

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java (hunk 4 above)
Comment (lewie) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java]: Oh yes, you are right. These Lines 376, 377 we do not need.  [changed](https://github.com/kgoderis/openhab2/pull/9/commits/8370ac5e940cb6abe81597e34df2b23ea3e82041)

File: addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java (hunk 1 above)
Comment (kgoderis) [reply addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java]: Yes, there is no other option. Code guidelines prescribe a non static logger however

**Pull request merged**